vendor/github.com/OpenBazaar/spvwallet/sortsignsend.go

```diff
@@ -629,6 +629,66 @@ func (w *SPVWallet) buildTx(amount int64, addr btc.Address, feeLevel wallet.FeeL
 	return authoredTx.Tx, nil
 }
 
+func NewUnsignedTransaction(outputs []*wire.TxOut, feePerKb btc.Amount, fetchInputs txauthor.InputSource, fetchChange txauthor.ChangeSource) (*txauthor.AuthoredTx, error) {
+
+	var targetAmount btc.Amount
+	for _, txOut := range outputs {
+		targetAmount += btc.Amount(txOut.Value)
+	}
+
+	estimatedSize := EstimateSerializeSize(1, outputs, true, P2PKH)
+	targetFee := txrules.FeeForSerializeSize(feePerKb, estimatedSize)
+
+	for {
```

Comment:
It seems like this could be done without a potentially infinite loop. It seems that the call to `fetchInputs` will only ever get two different inputs and should only return two different values.

Reply:
This is how it is written in btcd. I'll probably keep it for now and consider refactoring later.